chaincode/compute_plan.go

```diff
@@ -311,13 +316,116 @@ func getComputePlan(db *LedgerDB, key string) (resp outputComputePlan, err error
 		}
 	}
 
+	keys := []string{}
+	keys = append(keys, tupleKeys...)
+	keys = append(keys, testtupleKeys...)
+
+	status, err := getComputePlanStatus(db, keys)
+	if err != nil {
+		return
+	}
+
 	resp = outputComputePlan{
 		ComputePlanID:           key,
 		ObjectiveKey:            objectiveKey,
 		TraintupleKeys:          traintupleKeys,
 		CompositeTraintupleKeys: compositeTraintupleKeys,
 		AggregatetupleKeys:      aggregatetupleKeys,
 		TesttupleKeys:           testtupleKeys,
+		Status:                  status,
 	}
 	return
 }
+
+func getComputePlanStatusByComputePlanID(db *LedgerDB, computePlanID string) (status string, err error) {
+	computePlan, err := getComputePlan(db, computePlanID)
+	if err != nil {
+		return "", err
+	}
+
+	keys := []string{}
+	keys = append(keys, computePlan.TraintupleKeys...)
+	keys = append(keys, computePlan.CompositeTraintupleKeys...)
+	keys = append(keys, computePlan.AggregatetupleKeys...)
+	keys = append(keys, computePlan.TesttupleKeys...)
+
+	return getComputePlanStatus(db, keys)
```

Comment:
Now that I've read it like that I tend to think that `getComputePlan` should perhaps take the compute plan directly as argument and we should reconstruct this list in the function. What do you think @inalgnu

Reply:
@thibaultrobert that make sens, updated